panther_analysis_tool/cmd/configsdk_test.py

```diff
@@ -0,0 +1,280 @@
+import argparse
+import base64
+import enum
+import json
+import logging
+from collections import defaultdict
+from functools import reduce
+from typing import Tuple, Final, Dict, List, Any
+
+from panther_analysis_tool.cmd import config_utils
+
+
+class DetectionType(enum.Enum):
+    UNKNOWN = 1
+    RULE = 2
+    POLICY = 3
+    SCHEDULED_RULE = 4
+
+
+class Filter:
+    _PATH_TO_FILTER_SRC: Final = ['d', 'func', 'src']
+    _PATH_TO_FILTER_NAME: Final = ['d', 'func', 'name']
+
+    def __init__(self, _filter: Dict):
+        self.src = _deep_get(_filter, self._PATH_TO_FILTER_SRC, 'No source found')
+        self.name = _deep_get(_filter, self._PATH_TO_FILTER_NAME, 'No name found')
+
+    def get_code(self) -> str:
+        return base64.standard_b64decode(self.src).decode('utf8')
+
+    def get_name(self) -> str:
+        return self.name
+
+
+class UnitTest:
+    def __init__(self, test: Dict):
+        self.origin: str = _deep_get(test, ['o', 'name'], 'No origin found')
+        self.data: Dict = json.loads(_deep_get(test, ['d', 'data'], '{}'))
+        self.name: str = _deep_get(test, ['d', 'name'], 'No name found')
+        self.expect_match: bool = bool(_deep_get(test, ['d', 'expect_match'], True))
+        self.fail_reasons: List[str] = []  # only used if test failed
+
+    def get_prg(self, filters: List[Filter], detection_type: DetectionType) -> str:
+        prg = '_filters = [] \n\n'
+        for filt in filters:
+            prg += f'{filt.get_code()} \n\n'
+            prg += f'_filters.append({filt.get_name()}) \n\n'
+
+        # flip the return vals if it is a policy
+        match_val = False
+        if detection_type is DetectionType.POLICY:
+            match_val = True
+        no_match_val = not match_val
+
+        prg += 'def _execute(event):\n'
```

Comment:
You're not building an abstract syntax tree from scratch? casual

Reply:
this is interesting though, so you're basically building out the template defined [here](https://github.com/panther-labs/panther-enterprise/blob/master/cmd/devtools/pantherconfig/configsdk/apiversions/2022-08-17/detection_rule.py.tmpl#L19-L30)


how can we keep these in sync? would a comment suffice?

Reply:
Yes exactly. I could add a comment. Idk if it is an issue though for this open source to have a comment relating to our closed source code.

I also don't think it will be an issue since I would imagine once we set these they will never change.

Also the internal template is to support old detection running. Once we switch over completely to the config I think that template will go away

Reply:
good point, maybe we should reference this in panther-enterprise

But yeah, hopefully this never changes.